### PR TITLE
feat(diagnostic): добавлена диагностика небезопасного использования метода НайтиПоКоду()

### LINKED ISSUE #3475: [NEW] Найти по коду для с справочников
## Описание проблемы, ошибки, которую надо диагностировать

Использование короткой формы `Справочник.НайтиПоКоду(Код);`  
Для Справочника без контроля уникальности или с серией кодов не по всему справочнику может приводить к неожиданным результатам

### Тип [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [ ] :ant: Ошибка
* [ ] :cop: Уязвимость
* [ ] :guardsman: Потенциальная уязвимость
* [x] :poop: Качество кода
* [ ] :trollface: Другое

### Важность [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [ ] :broken_heart: Блокирующая / Blocker
* [ ] :heart: Критическая / Critical
* [ ] :yellow_heart: Важная / Major
* [ x] :blue_heart: Незначительная / Minor
* [ ] :green_heart: Информационная / Info
* [ ] :revolving_hearts: Другое

### Тэги [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTag)

* [ ] `STANDARD` - "Нарушение стандартов 1С"
* [ ] `LOCKINOS` - "Не будет работать в другой ОС"
* [ ] `SQL` - "Проблема с запросом"
* [ ] `PERFORMANCE` - "Проблема производительности"
* [ ] `BRAINOVERLOAD` - "Непонятный код"
* [ ] `BADPRACTICE` - "Плохая практика программирования"
* [ ] `CLUMSY` - "Излишние действия"
* [x] `DESIGN` - "Ошибка в проектировании"
* [x] `SUSPICIOUS` - "Подозрительный код"
* [ ] `UNPREDICTABLE` - "Непредсказуемо работающий код"
* [ ] `DEPRECATED` - "Устаревшая функциональность"
* [ ] `ERROR` - "Ошибочная конструкция"
* [ ] `LOCALIZE` - "Проблемы локализации"

### PR DESCRIPTION
**Pull request by johnnyshut**
## Описание

Добавлена диагностика `UnsafeFindByCodeDiagnostic` для обнаружения небезопасного использования метода `НайтиПоКоду()` (или `FindByCode()`) для справочников с отключенным контролем уникальности кода или включенными сериями кодов не по всему справочнику. Добавлена документация на русском и английском языках, тесты и метаданные для тестовых fixture. Проведен рефакторинг кода диагностики с добавлением тестов для граничных случаев.

## Связанные задачи

Closes #3475

### Общие

- [x] Ветка PR обновлена из develop
- [x] Отладочные, закомментированные и прочие, не имеющие смысла участки кода удалены
- [x] Изменения покрыты тестами
- [x] Обязательные действия перед коммитом выполнены (запускал команду `gradlew check licenseFormat updateJsonSchema`)

### Для диагностик

- [x] Описание диагностики заполнено для обоих языков (присутствуют файлы для обоих языков, для русского заполнено все подробно, перевод на английский можно опустить)

## Дополнительно

Обновлена зависимость mdclasses до версии 0.16.1-rc.2.

## Summary by CodeRabbit

* **New Features**
  * Added a diagnostic that detects unsafe use of FindByCode()/НайтиПоКоду for catalogs, charts of characteristic types, and charts of accounts with localized messages.

* **Documentation**
  * Added English and Russian diagnostic docs with invalid/valid examples and sources.

* **Tests**
  * Added unit tests covering safe, unsafe, and edge-case scenarios.

* **Chores**
  * Updated dependency version and added a configuration parameter/schema to enable/configure the diagnostic.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    var catalog = findCatalog(configuration, mdoRef);
+
+    try {
+      if (isUnsafeCatalogUsage(catalog)) {
+        diagnosticStorage.addDiagnostic(methodName);
+      }
+    } catch (IllegalStateException | IllegalAccessException | java.lang.reflect.InvocationTargetException e) {
+      // Ignore errors
+    }
+  }
```
[the remark this answers is not in the document]
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Реализовал

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    
+    if (!(codeSeries instanceof Enum<?> enumValue)) {
+      // Если codeSeries не является enum, считаем использование безопасным
+      return false;
+    }
+    
+    boolean checkUnique = (Boolean) checkUniqueMethod.invoke(catalog);
+    String codeSeriesName = enumValue.name();
+    return !checkUnique || !WHOLE_CATALOG_ENUM_VALUE.equals(codeSeriesName);
+  }
```
[the remark this answers is not in the document]
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Исправил

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+@DiagnosticMetadata(
+  type = DiagnosticType.CODE_SMELL,
+  severity = DiagnosticSeverity.MAJOR,
+  scope = DiagnosticScope.BSL,
+  minutesToFix = 5,
+  tags = {
+    DiagnosticTag.STANDARD,
+    DiagnosticTag.BADPRACTICE,
+    DiagnosticTag.UNPREDICTABLE
+  }
```
[the remark this answers is not in the document]
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java:78]: Сделал в следующем коммите

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    return !checkUnique || !WHOLE_CATALOG_ENUM_VALUE.equals(codeSeriesName);
+  }
+
+  /**
+   * Находит методы для проверки свойств справочника через рефлексию.
+   *
+   * @param catalog справочник для поиска методов
+   * @return объект с найденными методами; поля могут быть null, если соответствующие методы не найдены
+   */
+  private static CatalogMethods findCatalogMethods(Catalog catalog) {
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: предлагаю уйти от магии с рефлексией, а использовать явные касты с учетом нужных типов МДО

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/utils/MdoRefBuilder.java [hunk 5]
```diff
@@ -46,7 +46,7 @@ public class MdoRefBuilder {
   /**
    * Для оптимизации хранения одинаковых строк
    */
-  private final StringInterner stringInterner = new StringInterner();
+  private static final StringInterner stringInterner = new StringInterner();
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/utils/MdoRefBuilder.java]: убрать все статики - это lombok utility class

Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/utils/MdoRefBuilder.java]: *все добавленные static

File: src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/en/UnsafeFindByCode.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
+# Unsafe FindByCode() method usage
```
Comment (theshadowco) [inline src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/en/UnsafeFindByCode.md]: этот файл зачем?

File: src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/ru/UnsafeFindByCode.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# Небезопасное использование метода НайтиПоКоду()
```
Comment (theshadowco) [inline src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/ru/UnsafeFindByCode.md]: этот файл зачем?

File: src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/en/UnsafeFindByCode.md (hunk 6 above)
Comment (johnnyshut) [reply src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/en/UnsafeFindByCode.md]: рука дрогнула

File: src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/ru/UnsafeFindByCode.md (hunk 7 above)
Comment (johnnyshut) [reply src/main/resources/com/github/_1c_syntax/bsl/languageserver/diagnostics/ru/UnsafeFindByCode.md]: рука дрогнула

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+    return super.visitComplexIdentifier(ctx);
+  }
+
+  /**
+   * Проверяет, является ли вызов методом FindByCode/НайтиПоКоду и обрабатывает его.
+   *
+   * @param ctx контекст вызова метода
+   * @param mdoRef ссылка на объект метаданных
+   */
+  private void checkFindByCodeMethod(org.antlr.v4.runtime.ParserRuleContext ctx, String mdoRef) {
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: зачем полный тип? вроде конфликтов импортов не вижу

Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: укажи BSLParser.ComplexIdentifierContext

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+  /**
+   * Находит справочник по ссылке на объект метаданных.
+   *
+   * @param configuration конфигурация для поиска
+   * @param mdoRef ссылка на объект метаданных в формате "Catalog.ИмяКаталога"
+   * @return найденный справочник
+   * @throws java.util.NoSuchElementException если справочник не найден
+   */
+  private static Catalog findCatalog(CF configuration, String mdoRef) {
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: у Configuration есть методы поиска по mdoref и прочим полям

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+    boolean checkUnique = catalog.isCheckUnique();
+    Object codeSeries = catalog.getCodeSeries();
+    
+    if (!(codeSeries instanceof Enum<?>)) {
+      return false;
+    }
+    
+    Enum<?> enumValue = (Enum<?>) codeSeries;
+    String codeSeriesName = enumValue.name();
+    return !checkUnique || !WHOLE_CATALOG_ENUM_VALUE.equals(codeSeriesName);
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: метод getCodeSeries вернет нужный тип, зоначение проверяется обычным сравнением...

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 8 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Сделал

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 9 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Сделал

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 10 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Изменил

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  /**
+   * Проверяет, является ли вызов методом FindByCode/НайтиПоКоду и обрабатывает его.
+   *
+   * @param ctx контекст вызова метода
+   * @param mdoRef ссылка на объект метаданных
+   */
+  private void checkFindByCodeMethod(BSLParser.ComplexIdentifierContext ctx, String mdoRef) {
+    Methods.getMethodName(ctx).ifPresent((org.antlr.v4.runtime.Token methodName) -> {
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: тип надо исправить... и далее по файлу

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+    var configuration = documentContext.getServerContext().getConfiguration();
+
+    try {
+      var catalog = findCatalog(configuration, mdoRef);
+      if (isUnsafeCatalogUsage(catalog)) {
+        diagnosticStorage.addDiagnostic(methodName);
+      }
+    } catch (IllegalStateException 
+      | java.util.NoSuchElementException 
+      | ClassCastException e
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: эксепшенов больше нет, т.к. отказался от рефлексии

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+   * Находит справочник по ссылке на объект метаданных.
+   *
+   * @param configuration конфигурация для поиска
+   * @param mdoRef ссылка на объект метаданных в формате "Catalog.ИмяКаталога"
+   * @return найденный справочник
+   * @throws java.util.NoSuchElementException если справочник не найден
+   */
+  private static Catalog findCatalog(CF configuration, String mdoRef) {
+    var catalogName = mdoRef.substring(CATALOG_PREFIX.length());
+    var mdoReference = MdoReference.create(MDOType.CATALOG, catalogName);
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: MdoReference.create по строке полного имени (mdoRef) работает

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+  }
+)
+public class UnsafeFindByCodeDiagnostic extends AbstractVisitorDiagnostic {
+
+  private static final Pattern METHOD_NAME_PATTERN = CaseInsensitivePattern.compile(
+    "^(НайтиПоКоду|FindByCode)$"
+  );
+
+  private static final String CATALOG_PREFIX = "Catalog.";
+  private static final String WHOLE_CATALOG_ENUM_VALUE = "WHOLE_CATALOG";
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: здесь не надо строкой, сразу значение перечисления использовать

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+   * - серии кодов установлены не для всего справочника
+   *
+   * @param catalog справочник для проверки
+   * @return true, если использование небезопасно; false, если использование безопасно
+   */
+  private static boolean isUnsafeCatalogUsage(Catalog catalog) {
+    Enum<?> codeSeries = catalog.getCodeSeries();
+    String codeSeriesName = codeSeries.name();
+    boolean checkUnique = catalog.isCheckUnique();
+    return !checkUnique || !WHOLE_CATALOG_ENUM_VALUE.equals(codeSeriesName);
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: весь метод превращается в `return !catalog.isCheckUnique() ||  catalog.getCodeSeries() == WHOLE_CATALOG_ENUM_VALUE`

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+   * и серии кодов установлены для всего справочника. Если хотя бы одно из условий
+   * не выполняется, добавляется диагностика.
+   * <p>
+   * Ошибки рефлексии и поиска методов игнорируются, диагностика не добавляется.
+   *
+   * @param mdoRef ссылка на объект метаданных в формате "Catalog.ИмяКаталога"
+   * @param methodName токен с именем метода FindByCode/НайтиПоКоду
+   */
+  private void checkCatalog(String mdoRef, org.antlr.v4.runtime.Token methodName) {
+    if (!mdoRef.startsWith(CATALOG_PREFIX)) {
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: а проверять для ПВХ и плана счетов не будешь?

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+   *
+   * @param configuration конфигурация для поиска
+   * @param mdoRef ссылка на объект метаданных в формате "Catalog.ИмяКаталога"
+   * @return найденный справочник
+   * @throws java.util.NoSuchElementException если справочник не найден
+   */
+  private static Catalog findCatalog(CF configuration, String mdoRef) {
+    var catalogName = mdoRef.substring(CATALOG_PREFIX.length());
+    var mdoReference = MdoReference.create(MDOType.CATALOG, catalogName);
+    Optional<MD> catalogOptional = configuration.findChild(mdoReference);
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: кстати, сюда ведь тоже можно сразу строку передавать, из нее mdoref сам mdclass соберет

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 16 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Забыл. Сделаю

Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Сделал

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 15 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Исправил, спасибо

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 17 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Сделал

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 11 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: То есть перенести `org.antlr.v4.runtime.Token` в импорт и оставить 
`Methods.getMethodName(ctx).ifPresent((Token methodName) -> {`
?

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 14 above)
Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Реализовал через перечисление

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java (hunk 11 above)
Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: ну да

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 18]
```diff
@@ -0,0 +1,327 @@
...
+   * Если объект метаданных не найден в конфигурации или {@code mdoRef} не начинается с известных префиксов
+   * (например, для документов), диагностика не добавляется.
+   *
+   * @param mdoRef ссылка на объект метаданных в формате "Catalog.ИмяКаталога",
+   *               "ChartOfCharacteristicTypes.ИмяПлана" или "ChartOfAccounts.ИмяПлана"
+   * @param methodName токен с именем метода FindByCode/НайтиПоКоду
+   */
+  private void checkMetadataObject(String mdoRef, Token methodName) {
+    var configuration = documentContext.getServerContext().getConfiguration();
+
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java:155]: как-то сложно. 
посмотри такой вариант:
1. по mdoRef получаешь объект `var mdoOpt =configuration.findChild(mdoRef) // тут optional` 
2. Если optional пустой, то возврат
3. иначе `var mdo = mdoOpt.get()` получил MDO
4. у каждого mdo есть метод, возвращающий MdoType. По нему ты проверяешь
  - если Catalog - проверка каталога
  - есть ChartOfCharacteristicTypes - проверка пвх
  - ....
  - иначе - возврат
5. по результатам п.4 получил флаг, воторый возвращаешь в метод checkFindMethod, в котором просто вызываешь `addDiagnostic(ctx)`

будет в разы короче и стабильнее

Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java:155]: 5. можно и на токен метода, если хочется именно его выделить

Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java:155]: Код стал намного четче! Спасибо!

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+   */
+  private void addDiagnosticOnCommentLine(Token methodName) {
+    int methodLineANTLR = methodName.getLine();
+    int commentLineLSP = methodLineANTLR - COMMENT_LINE_OFFSET;
+    if (commentLineLSP >= 0) {
+      diagnosticStorage.addDiagnostic(commentLineLSP, 0, commentLineLSP, Integer.MAX_VALUE);
+    } else {
+      diagnosticStorage.addDiagnostic(methodName);
+    }
+  }
```
Comment (theshadowco) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: вопрос: почему так? 

Вот строка у тебя `а = справочники.мойсправочник.найтипокоду(7)`. Ты вышаешь замечание на некую предыдущую строку комментария. Обычно замечание вешается на само место, не соответствующее требованиям, т.е. 
 - либо на `справочники.мойсправочник.найтипокоду(7)`
 - либо на `справочники.мойсправочник.найтипокоду`
 - либо на `найтипокоду`

Пояснишь свой выбор?

Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Потому что тест возвращал на одну строку раньше, чем находился код (там где комментарий). Подумал, что это потому что комментарии писал сверху, а не справа от кода. И сделал такую заглушку, чтобы номера были как во всех тестах

Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: Это особенность нумерации, так везде

Comment (johnnyshut) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/UnsafeFindByCodeDiagnostic.java]: По моему понял Тебя, можно проверять
